### ISSUE TITLE
Feature request: ability to position the overall column to the left?

### ISSUE THREAD

**Issue by Beduiz** [benjaminrich/table1#88]
Dear Benjamin,

It would be great if there was an option to position the overall column, just like the extra.col can be positioned with extra.col.pos. I tried with overall.pos = 1, but it didnt respond, and I also couldnt find this in the documentation, so I assume it is not a feature as of now?

Kind regards
Eric

**Comment by EpiPenny** [comment]
Thanks for asking this.  I am interested in this option too.

**Comment by g-antonello** [comment]
Same here

**Comment by EpiPenny** [comment]
Hi Everyone,

I was able to figure out how to do this based on an example posted here:

https://cran.r-project.org/web/packages/table1/vignettes/table1-examples.html

It's the last modification demonstrated in Example 1.

**Comment by benjaminrich** [comment]
@EpiPenny: you are quite right, this is a perfectly valid solution. But I think that what users would like is an easier way to do this. I've pushed a solution in b27cd6d. Basically, instead of just specifying a label like `overall="Overall"`, if you want it to be on the left you give it the name "left", like this: `overall=c(left="Overall")`.

**Comment by EpiPenny** [comment]
Hi,

Thanks for responding to this question!!!  That definitely seems like a more straightforward solution.  However, when I tried it, the overall column still printed on the right.  Below is my code.  Did I misunderstand how to implement your suggestion, or might the problem be related to using an older version of the package?

table1(~ earlycare_f + mage5yr_fa + mraceeth_f + smoke_f |
                   preterm_f,
                 data = birthsfin,
                 overall=c(left="All Births"),
                 render.categorical = "Freq (PCTnoNA%)",
                 render.missing = "Freq",
                 big.mark = ",",
                 #This makes caption bold in html but not not word
                 caption = "<b>Table 1. Characteristics of a Cohort of U.S., Singleton, Live Births by Preterm Birth Status Based on the 2006 U.S. Birth Certificate Data\u{1D43}</b>",
                 #This works for viewer, but not for exporting to word (use add_footer_lines for Word)
                 footnote = c("\u{1D43}Only births reported on the 2003 revised U.S. Standard Birth Certificate were eligible",
                           "\u{1D47}Preterm birth includes live births that occurred at gestational ages 21-36 weeks inclusive")
  )

Thanks,
Penny

From: Benjamin Rich ***@***.***>
Sent: Monday, January 2, 2023 2:08 PM
To: benjaminrich/table1 ***@***.***>
Cc: Howards, Penelope ***@***.***>; Mention ***@***.***>
Subject: [External] Re: [benjaminrich/table1] Feature request: ability to position the overall column to the left? (Issue #88)


@EpiPenny<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FEpiPenny&data=05%7C01%7Cphoward%40emory.edu%7C8310c0fd38ca4561527d08daecf4a101%7Ce004fb9cb0a4424fbcd0322606d5df38%7C0%7C0%7C638082832673531725%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Cqal7dk7Blr%2FRXc9V13Ne3KYpLxHJodKX%2BM16P%2Fd9JQ%3D&reserved=0>: you are quite right, this is a perfectly valid solution. But I think that what users would like is an easier way to do this. I've pushed a solution in b27cd6d<https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbenjaminrich%2Ftable1%2Fcommit%2Fb27cd6d616f56e7bc26003a1387adaef540f37d3&data=05%7C01%7Cphoward%40emory.edu%7C8310c0fd38ca4561527d08daecf4a101%7Ce004fb9cb0a4424fbcd0322606d5df38%7C0%7C0%7C638082832673531725%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=7b%2BncG6K6V1n%2BIkcOOQ3hpWtbU3T5V5h4kWt2jvlfKU%3D&reserved=0>. Basically, instead of just specifying a label like overall="Overall", if you want it to be on the left you give it the name "left", like this: overall=c(left="Overall").

-

**Comment by EpiPenny** [comment]
Hi,

Thanks for responding to this question!!!  That definitely seems like a more straightforward solution.  However, when I tried it, the overall column still printed on the right.  Below is my code.  Did I misunderstand how to implement your suggestion, or might the problem be related to using an older version of the package?

```
#Table 1
table1(~ earlycare_f + mage5yr_fa + mraceeth_f + smoke_f | 
                   preterm_f, 
                 data = birthsfin, 
                 overall=c(left="All Births"),
                 render.categorical = "Freq (PCTnoNA%)", 
                 render.missing = "Freq",
                 big.mark = ",",
                 #This makes caption bold in html but not not word
                 caption = "<b>Table 1. Characteristics of a Cohort of U.S., Singleton, Live Births by Preterm Birth Status Based on the 2006 U.S. Birth Certificate Data\u{1D43}</b>",
                 #This works for viewer, but not for exporting to word (use add_footer_lines for Word)
                 footnote = c("\u{1D43}Only births reported on the 2003 revised U.S. Standard Birth Certificate were eligible",
                           "\u{1D47}Preterm birth includes live births that occurred at gestational ages 21-36 weeks inclusive")
  )
```

Thanks!!!

**Comment by benjaminrich** [comment]
Hi @EpiPenny. In order to try this out prior to the CRAN release, you need to install the "development" version using the `remotes` package:

```r
remotes::install_github("benjaminrich/table1")
```

Let me know if that works. Thanks.

**Comment by benjaminrich** [comment]
This has been implemented in CRAN release 1.4.3.

**Comment by EpiPenny** [comment]
Hi,

I tried it, and it worked perfectly!

Thanks so much for addressing this!
